synchronization/src/cyface/java/de/cyface/synchronization/CyfaceSyncAdapter.java

```diff
@@ -128,31 +127,35 @@ public void onPerformSync(final @NonNull Account account, Bundle extras, String
         Cursor unsyncedMeasurementsCursor = null;
         try {
             unsyncedMeasurementsCursor = MeasurementContentProviderClient.loadSyncableMeasurements(provider, authority);
-            boolean atLeastOneMeasurementExists = unsyncedMeasurementsCursor.moveToNext();
+            final boolean atLeastOneMeasurementExists = unsyncedMeasurementsCursor.moveToNext();
             if (!atLeastOneMeasurementExists) {
-                Log.i(TAG, "Unable to sync data: " + SYNC_ERROR_MESSAGE_NO_UN_SYNCED_DATA);
+                Log.i(TAG, "Unable to sync data: no unsynchronized data");
                 return;
             }
             notifySyncStarted(countUnsyncedDataPoints(provider, unsyncedMeasurementsCursor, authority));
 
-            GeoLocationJsonMapper geoLocationJsonMapper = new GeoLocationJsonMapper();
-            AccelerationJsonMapper accelerationJsonMapper = new AccelerationJsonMapper();
-            RotationJsonMapper rotationJsonMapper = new RotationJsonMapper();
-            DirectionJsonMapper directionJsonMapper = new DirectionJsonMapper();
+            final GeoLocationJsonMapper geoLocationJsonMapper = new GeoLocationJsonMapper();
+            final AccelerationJsonMapper accelerationJsonMapper = new AccelerationJsonMapper();
+            final RotationJsonMapper rotationJsonMapper = new RotationJsonMapper();
+            final DirectionJsonMapper directionJsonMapper = new DirectionJsonMapper();
 
             // The cursor is reset to initial position (i.e. 0) by countUnsyncedDataPoints
             do {
```

Comment:
This method is way too long. A good start would be to move everything in this do-while loop into its own private method, since it describes the synchronization of exactly one measurement.